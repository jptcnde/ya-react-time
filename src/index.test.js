import React   from 'react';
import { shallow } from 'enzyme';

import Time from './index';

describe('Time', () => {
  const date = new Date('1995-12-17 11:39:09');

  it('should render', () => {
    const actual = shallow(<Time>{date}</Time>);
    expect(actual).toMatchSnapshot();
  });

  it('should format specified date', () => {
    const wrapper = shallow(<Time format="YYYY/MM/DD">{date}</Time>);
    const expected = '1995/12/17';
    expect(wrapper.text()).toBe(expected);
    expect(wrapper).toMatchSnapshot();
  });

  it('should filter', () => {
    const wrapper = shallow(
      <Time
        format="HH:mm:ss A"
        filter={d => d.toLowerCase()}
      >
        {date}
      </Time>
    );
    const expected = '11:39:09 am';

    expect(wrapper.text()).toBe(expected);
    expect(wrapper).toMatchSnapshot();
  });

  it('should format with Time "format" helper', () => {
    const expected = '1995/12/17';

    expect(Time.format(date, 'YYYY/MM/DD')).toBe(expected);
  });

  it('should parse with Time "parse" helper', () => {
    const expected = 'Invalid Date';
    expect(Time.parse(date)).not.toBe(expected);
  });
});