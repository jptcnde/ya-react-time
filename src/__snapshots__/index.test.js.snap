// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time should filter 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Time
    component="span"
    filter={[Function]}
    format="HH:mm:ss A"
    value=""
  >
    1995-12-17T03:39:09.000Z
  </Time>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "11:39:09 am",
      "value": "",
    },
    "ref": null,
    "rendered": "11:39:09 am",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "11:39:09 am",
        "value": "",
      },
      "ref": null,
      "rendered": "11:39:09 am",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Time should format specified date 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Time
    component="span"
    filter={[Function]}
    format="YYYY/MM/DD"
    value=""
  >
    1995-12-17T03:39:09.000Z
  </Time>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "1995/12/17",
      "value": "",
    },
    "ref": null,
    "rendered": "1995/12/17",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1995/12/17",
        "value": "",
      },
      "ref": null,
      "rendered": "1995/12/17",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Time should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Time
    component="span"
    filter={[Function]}
    format="YYYY-MM-DDTHH:mm:ss"
    value=""
  >
    1995-12-17T03:39:09.000Z
  </Time>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "1995-12-17T11:39:09",
      "value": "",
    },
    "ref": null,
    "rendered": "1995-12-17T11:39:09",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1995-12-17T11:39:09",
        "value": "",
      },
      "ref": null,
      "rendered": "1995-12-17T11:39:09",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
